src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -47,6 +44,8 @@ const signEventOrPrompt = async (message: MessageSignEvent) => {
       }
     }
 
+    if (!event.id) event.id = nostr.getEventHash(event);
```

Comment:
I think we should always set the pubkey and id. The pubkey must be the pubkey of the event creator, and the ID can only be generated once all fields including the pubkey are set on the event.

**important** the pubkey must be set in order to calculate the event ID. When you moved them you swapped the order around which will break `getEventHash`.

Reply:
nice catch! after we move this to after the validation we swapped pubkey and event.id order, you are right, the event id should be after everything else is part of the event object

Reply:
as for the "always set the pubkey and id" the idea then is to always override whatever event we receive to sign with the pubkey from alby, makes sense

Reply:
I think so. Otherwise we need to validate them, and the only way to validate them is to compare against the generated values, so we might as well overwrite them?